services/checks/registry.go

```diff
@@ -0,0 +1,76 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package checks
+
+import (
+	"sync"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/percona/pmm/api/alertmanager/ammodels"
+)
+
+// registry stores alerts and delay information by IDs.
+type registry struct {
+	rw     sync.RWMutex
+	alerts map[string]ammodels.PostableAlert
+	nowF   func() time.Time // for tests
+}
+
+// newRegistry creates a new registry.
+func newRegistry() *registry {
+	return &registry{
+		alerts: make(map[string]ammodels.PostableAlert),
+		nowF:   time.Now,
+	}
+}
+
+// createAlert creates alert from given AlertParams
+func (r *registry) createAlert(labels, annotations map[string]string, alertTTL time.Duration) ammodels.PostableAlert {
```

Comment:
That should be a function, not a method. Just remove receiver

Reply:
if you look further down the method I need access to `nowF` which is a field of registry that is why it is a method. Do you still want me to remove the receiver and add the endsAt field in the collect method? @AlekSi

Reply:
Nope, I stand corrected